### PR TITLE
Property access reuse

### PR DESCRIPTION
**Pull request by Koc**
Reuse existing property access instance for performance improvement.

Reasons:
1. starting from Symfony v2.3.35 there is internal cache for this component (cache inside request) https://github.com/symfony/symfony/commit/284dc75796b693562e2c091fd770c4a04c179a2b . Cache working using class property and we should use same instance to get it works properly.
2. there is unmerged yet PR that adds cache outside request https://github.com/symfony/symfony/pull/16838 . For getting it working we should use configured instance with passed cache layer inside it instead of creating new one.
3. in two classes you instanciate property accessor inside foreach.

### CODE REVIEW

File: Admin/Admin.php [hunk 1]
```diff
@@ -945,7 +945,7 @@ protected function buildForm()
         if ($this->isChild() && $this->getParentAssociationMapping()) {
             $parent = $this->getParent()->getObject($this->request->get($this->getParent()->getIdParameter()));
 
-            $propertyAccessor = PropertyAccess::createPropertyAccessor();
+            $propertyAccessor = $this->getConfigurationPool()->getContainer()->get('property_access');
```
Comment (Koc) [inline Admin/Admin.php]: We can replace it with setter, but who should call this setter?

**Comment by Koc** [comment]
Looks like tests are finally pass. Please review.

**Comment by Koc** [comment]
@rande @OskarStark @Soullivaneuh can anybody says anything about this PR?

**Comment by OskarStark** [comment]
Status: Reviewed

Looks good to me in general, but I'm not very familar with this.

@core23 can you please have a look?

@rande I'm :+1: for merging this

@Koc thank you for your work!

**Comment by core23** [comment]
Looks okay to me, but I am not that deep in the caching stuff.

**Comment by OskarStark** [comment]
Thank you @core23

**Comment by Koc** [comment]
So does anything block this?

File: Admin/Pool.php [hunk 2]
```diff
@@ -23,7 +25,7 @@ class Pool
     /**
      * @var ContainerInterface|null
```
Comment (OskarStark) [inline Admin/Pool.php]: php doc wrong, if not null

File: Form/ChoiceList/ModelChoiceList.php [hunk 3]
```diff
@@ -146,14 +153,13 @@ protected function load($choices)
...
-                $propertyAccessor = PropertyAccess::createPropertyAccessor();
-                $value = $propertyAccessor->getValue($entity, $this->propertyPath);
+                $value = $this->propertyAccessor->getValue($entity, $this->propertyPath);
             } else {
                 // Otherwise expect a __toString() method in the entity
                 try {
                     $value = (string) $entity;
                 } catch (\Exception $e) {
-                    throw new RuntimeException(sprintf("Unable to convert the entity %s to String, entity must have a '__toString()' method defined", ClassUtils::getClass($entity)), 0, $e);
+                    throw new RuntimeException(sprintf('Unable to convert the entity "%s" to string, provide "property" option or implement "__toString()" method in your entity.', ClassUtils::getClass($entity)), 0, $e);
```
Comment (OskarStark) [inline Form/ChoiceList/ModelChoiceList.php]: can you linebreak this please?

**Comment by OskarStark** [comment]
@Koc after my comments i will merge this

**Comment by Koc** [comment]
@OskarStark PR rebased and I've reflected last changes, introduced in #3553. CS also have been fixed. Travis is green.

**Pull request merged**